### PR TITLE
TEK830A-15 Add logic for mimicking user

### PR DESCRIPTION
**Pull request by filipkauffeldt**
- Create a basic `Schedule` class that holds a lighting schedule.
- Create an abstract `Scheduler` base class, used for creating schedules from user data.
- Create a concrete `RandomScheduler` implementation that should produce a random schedule based on the user data. Not A very good solution. Is based very much upon Copilot code, but produces some type of result.

### CODE REVIEW

**Review by BooloCaptain — APPROVED** [review]
looke goode

File: model/model.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 
 class Model():
-    def __init__(self, data) -> None:
+    def __init__(self, data: str, scheduler: Scheduler) -> None:
         self.user_data = self.read_data(data)
-        self.schedule = self.update_schedule(self.user_data)
+        self.schedule: Schedule = self.update_schedule(self.user_data, scheduler)
 
     def mainloop(self):
+        pass
```
Comment (RezaAmanii) [inline model/model.py:12]: why you have "pass" in this method?

File: model/model.py [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
         #self.user_data = self.update_data()
```
Comment (RezaAmanii) [inline model/model.py:13]: do you need this line at all, because it is commented?

File: model/model.py [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
-        self.schedule = self.update_schedule(self.user_data)
+        # self.schedule = self.update_schedule(self.user_data,)
```
Comment (RezaAmanii) [inline model/model.py:14]: do you need this line at all, because it is commented?

**Review by RezaAmanii — APPROVED** [review]
I added some comments otherwise it looks good to me

File: model/model.py (hunk 3 above)
Comment (filipkauffeldt) [reply model/model.py:14]: Not sure if these are needed but not implemented yet

File: model/model.py [hunk 4]
```diff
@@ -1,15 +1,17 @@
 import json
 from typing import List
 from model.events.lamp_event import LampEvent
+from model.scheduler import *
```
Comment (Flaskhals68) [inline model/model.py]: Maybe state exactly what is imported from model.scheduler

**Pull request merged**